applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -1696,11 +1710,24 @@ void SmallStrainUPwDiffOrderElement::CalculateDerivativesOnInitialConfiguration(
     KRATOS_CATCH("")
 }
 
-void SmallStrainUPwDiffOrderElement::CalculateBMatrix(Matrix& rB, const Matrix& DNp_DX, const Vector& Np)
+void SmallStrainUPwDiffOrderElement::CalculateBMatrix(Matrix& rB, const Matrix& DNp_DX, const Vector& Np) const
 {
     rB = mpStressStatePolicy->CalculateBMatrix(DNp_DX, Np, this->GetGeometry());
 }
 
+std::vector<Matrix> SmallStrainUPwDiffOrderElement::CalculateBMatrices(
+    const Matrix& NContainer, const GeometryType::ShapeFunctionsGradientsType& DN_DXContainer) const
+{
+    std::vector<Matrix> result;
+    for (unsigned int GPoint = 0; GPoint < DN_DXContainer.size(); ++GPoint) {
+        Matrix b_matrix;
+        this->CalculateBMatrix(b_matrix, DN_DXContainer[GPoint], row(NContainer, GPoint));
```

Comment:
Would it be useful to return b_matrix from this->CalculateBMatrix to simplify this for loop? For example, then std::transform can be used instead of the for loop.

Reply:
Yes completely agree, changed it to 'return-based'. Unfortunately we cannot use std::transform, since the NContainer is a Matrix instead of a vector<vector<...>>. Therefore, we need the GPoint index now to get the right row from NContainer. But if you (or e.g. @avdgraaf) sees a way to do it, please let me know!